Mathlib/Data/Fintype/Basic.lean

```diff
@@ -227,6 +227,9 @@ theorem compl_insert : (insert a s)ᶜ = sᶜ.erase a := by
   simp only [not_or, mem_insert, iff_self_iff, mem_compl, mem_erase]
 #align finset.compl_insert Finset.compl_insert
 
+theorem insert_compl_insert (ha : a ∉ s) : insert a (insert a s)ᶜ = sᶜ := by
+  simp_rw [@eq_compl_comm _ _ s, compl_insert, compl_erase, compl_compl, erase_insert ha]
```

Comment:
I would drop this lemma since it is the combination of two much simpler rewrites:
```suggestion
  simp_rw [compl_insert, insert_erase (mem_compl.2 ha)]
```

Reply:
Thanks for the simpler proof.
I still think it's useful to keep, since you use 3 lemmas, and this is a simple statement that will probably occur more frequently when doing things with `insert` and `compl` (for manual rewriting - it is indeed not a good `simp` lemma).

Reply:
My point is that in all cases where you would be faced with this expression, you will first rewrite `compl_insert`. Then the proof is just `insert_erase (mem_compl.2 ha)`, which doesn't seem to be worth a lemma.

Reply:
Should `insert_erase` be a `simp` lemma so that `simp` applies all three steps?

Reply:
It really could, yeah.